### PR TITLE
[WIP] Compute rooms on splitted reservations

### PR DESCRIPTION
**Pull request by miguelpadin**
@DarioLodeiros, can you check this please ?

### CODE REVIEW

File: pms/models/pms_reservation_line.py [hunk 1]
```diff
@@ -91,37 +91,98 @@ def name_get(self):
...
+                    checkout=line.reservation_id.checkout,
+                    room_type_id=line.reservation_id.room_type_id.id,
+                    current_lines=line._origin.reservation_id.reservation_line_ids.ids,
                 )
-                if rooms_available:
-                    if preferred_room.id in rooms_available.ids:
-                        room_chosen = preferred_room
+
+                # if there is availability for the entire stay
+                if rooms_available and len(rooms_available) > 0:
```
Comment (DarioLodeiros) [inline pms/models/pms_reservation_line.py]: An empty recorset return False, then, its not necesary len() condition

File: pms/tests/test_pms_reservation.py [hunk 2]
```diff
@@ -30,6 +94,399 @@ def test_create_reservation(self):
...
+        +------------+------+------+------+----+----+----+
+        """
+
+        self.create_common_scenario()
+
+        r_test = self.env["pms.reservation"].create(
+            {
+                "pms_property_id": self.property.id,
+                "checkin": datetime.datetime.now(),
+                "checkout": datetime.datetime.now() + datetime.timedelta(days=3),
```
Comment (DarioLodeiros) [inline pms/tests/test_pms_reservation.py:121]: do we use fixed or relative data?

**Review by DarioLodeiros — COMMENTED** [review]
Great Job! It's sound really good! :D  
You must name the PR more in accordance with the User Story (Compute Rooms on splitted reservations for example?)  and take into account the comments detailed in the code.
I still haven't had time to take a look at the test cases you propose, but the code looks good

File: pms/tests/test_pms_reservation.py (hunk 2 above)
Comment (miguelpadin) [reply pms/tests/test_pms_reservation.py:121]: We use the _freezegun_ library in each test case (method) to set the current date to a faked date.
For the reservation checkins and checkouts we use dates adding days to the faked date.

File: pms/models/pms_reservation_line.py (hunk 1 above)
Comment (miguelpadin) [reply pms/models/pms_reservation_line.py]: ok

**Pull request merged**